File: src/libs/ThirdParty.CSSuppliers/DOTW/InjectedDOTWSettings.cs
﻿namespace ThirdParty.CSSuppliers.DOTW
{
    using Intuitive.Helpers.Extensions;
    using ThirdParty;
    using ThirdParty.Constants;
    using ThirdParty.Support;

    public class InjectedDOTWSettings : SettingsBase, IDOTWSettings
    {
        protected override string Source => ThirdParties.DOTW;

        public string CompanyCode(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("CompanyCode", tpAttributeSearch);
        }

        public string CustomerCountryCode(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("CustomerCountryCode", tpAttributeSearch);
        }

        public string LeadGuestNationality(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("LeadGuestNationality", tpAttributeSearch);
        }

        public int DefaultCurrencyID(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("DefaultCurrencyID", tpAttributeSearch).ToSafeInt();
        }

        public bool ExcludeDOTWThirdParties(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("ExcludeDOTWThirdParties", tpAttributeSearch).ToSafeBoolean();
        }

        public string Password(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("Password", tpAttributeSearch);
        }

        public bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("AllowCancellations", tpAttributeSearch).ToSafeBoolean();
        }

        public string RequestCurrencyID(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("RequestCurrencyID", tpAttributeSearch);
        }

        public bool UseGZip(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("UseGZip", tpAttributeSearch).ToSafeBoolean();
        }

        public int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool isMandatory)
        {
            return Get_Value("OffsetCancellationDays", tpAttributeSearch).ToSafeInt();
        }

        public bool SendTradeReference(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("SendTradeReference", tpAttributeSearch).ToSafeBoolean();
        }

        public string GenericURL(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("GenericURL", tpAttributeSearch);
        }

        public string ThreadedSearch(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("ThreadedSearch", tpAttributeSearch);
        }

        public bool UseMinimumSellingPrice(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("UseMinimumSellingPrice", tpAttributeSearch).ToSafeBoolean();
        }

        public string User(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("User", tpAttributeSearch);
        }

        public int Version(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("Version", tpAttributeSearch).ToSafeInt();
        }
    }
}

File: src/libs/ThirdParty.VBSuppliers/Properties/DOTW/InjectedDOTWSettings.vb
Imports Intuitive.Helpers.Extensions
Imports ThirdParty.Abstractions
Imports ThirdParty.Abstractions.Search.Settings
Imports ThirdParty.Abstractions.Support.SettingsSupport

Public Class InjectedDOTWSettings
    Implements IDOTWSettings

    Private ReadOnly configuration As ThirdPartyConfiguration

    Public Sub New(ByVal configuration As ThirdPartyConfiguration)
        Me.configuration = configuration
    End Sub


    Public ReadOnly Property CompanyCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.CompanyCode
        Get
            Return Get_Value("CompanyCode", configuration)
        End Get
    End Property

    Public ReadOnly Property CustomerCountryCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.CustomerCountryCode
        Get
            Return Get_Value("CustomerCountryCode", configuration)
        End Get
    End Property

    Public ReadOnly Property CustomerNationality(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.CustomerNationality
        Get
            Return Get_Value("CustomerNationality", configuration)
        End Get
    End Property

    Public ReadOnly Property DefaultCurrencyID(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer Implements IDOTWSettings.DefaultCurrencyID
        Get
            Return Get_Value("DefaultCurrencyID", configuration).ToSafeInt()
        End Get
    End Property

    Public ReadOnly Property ExcludeDOTWThirdParties(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IDOTWSettings.ExcludeDOTWThirdParties
        Get
            Return Get_Value("ExcludeDOTWThirdParties", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.Password
        Get
            Return Get_Value("Password", configuration)
        End Get
    End Property

    Public ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IDOTWSettings.AllowCancellations
        Get
            Return Get_Value("AllowCancellations", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property RequestCurrencyID(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.RequestCurrencyID
        Get
            Return Get_Value("RequestCurrencyID", configuration)
        End Get
    End Property

    Public ReadOnly Property UseGZip(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IDOTWSettings.UseGZip
        Get
            Return Get_Value("UseGZIP", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Integer Implements IDOTWSettings.OffsetCancellationDays
        Get
            Return Get_Value("OffsetCancellationDays", configuration).ToSafeInt()
        End Get
    End Property

    Public ReadOnly Property SendTradeReference(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IDOTWSettings.SendTradeReference
        Get
            Return Get_Value("SendTradeReference", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property ServerURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.ServerURL
        Get
            Return Get_Value("ServerURL", configuration)
        End Get
    End Property

    Public ReadOnly Property ThreadedSearch(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.ThreadedSearch
        Get
            Return Get_Value("ThreadedSearch", configuration)
        End Get
    End Property

    Public ReadOnly Property UseMinimumSellingPrice(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IDOTWSettings.UseMinimumSellingPrice
        Get
            Return Get_Value("UseMinimumSellingPrice", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property Username(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.Username
        Get
            Return Get_Value("Username", configuration)
        End Get
    End Property

    Public ReadOnly Property Version(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IDOTWSettings.Version
        Get
            Return Get_Value("Version", configuration)
        End Get
    End Property

End Class
